flocker/node/_loop.py

```diff
@@ -501,15 +523,23 @@ def got_local_state(local_state):
                 self.configuration, self.cluster_state, local_state
             )
             if isinstance(action, NoOp):
+                # If we have converged, we need to reset the sleep delay
+                # in case there were any incremental back offs while
+                # waiting to converge.
+                self._unconverged_sleep.reset_delay()
                 # We've converged, we can sleep for NoOp's sleep duration.
                 # We add some jitter so not all agents wake up at exactly
                 # the same time, to reduce load on system:
+                self._unconverged_sleep.reset_delay()
                 sleep_duration = _Sleep.with_jitter(
                     action.sleep.total_seconds())
             else:
                 # We're going to do some work, we should do another
-                # iteration quickly in case there's followup work:
-                sleep_duration = _UNCONVERGED_DELAY
+                # iteration, but chances are that if, for any reason,
+                # the backend is saturated, by looping too fast, we
+                # will only make things worse, so there is an incremental
+                # back off in the sleep interval.
+                sleep_duration = self._unconverged_sleep.sleep()
```

Comment:
:grey_question: I'm 50/50 on this. There is definitely something smarter we could do here.

The bad: It's only 6 hits for this to max out at the 400 seconds. I guess I don't have a good feel for how often this will or won't occur, It seems like it'd be kind of bad if there is a 400 second wait before any volume could be attached. Could we add logging and then run the benchmarking cluster with this to gather more data? A new blockdevice goes from non-existing->CREATE->non-manifest->ATTACH->attached_no_fs->CREATE_FS->attached->MOUNT->mounted. I think that means we're already at waiting 25 seconds if another configuration change comes sometime during that process.

The good: This does put a limit at how quickly we will call discover_state (and thus backend list volume calls). It's a shame for a single blockdevice between CREATE_FS and MOUNT we can't do a cheaper discover_state that just dos lsblkid, and not querying the cloud.

Hrm... I think my main concerns could probably be addressed by seeing the results of some benchmarking runs on this code.